core/src/main/scala/slog/LoggingContext.scala

```diff
@@ -0,0 +1,15 @@
+package slog
+
+trait LoggingContext[F[_]] {
+  def use[T: StructureEncoder](key: String,
+                               value: T): LoggingContext.LoggingBuilder[F]
+}
+
+object LoggingContext {
+  def apply[F[_]](implicit ev: LoggingContext[F]): LoggingContext[F] = ev
+
+  trait LoggingBuilder[F[_]] {
+    def scoped[T](fv: F[T]): F[T]
+    def use[T: StructureEncoder](key: String, value: T): LoggingBuilder[F]
```

Comment:
so we are giving up on context propagation being orthogonal to structured logging?

Reply:
I don't see a way how can structured logging be implemented without some sort of context propagation. But they can live next to each other just fine.

Reply:
@tomasherman I think this library should be focused on logging only, it shouldn't concert with context propagation in general IMHO. at least not in the beginning

:thinking: 
thinking out loud: If we wanted to have general context propagation mechanisms, we probably should have started with that, and build logging (and what else? tracing? correlationID propagation?) on to of it.